internal/services/compute/capacity_reservation_group_resource.go

```diff
@@ -0,0 +1,163 @@
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
+		}
+	}
+	if !utils.ResponseWasNotFound(existing.Response) {
+		return tf.ImportAsExistsError("azurerm_capacity_reservation_group", id.ID())
+	}
+
+	parameters := compute.CapacityReservationGroup{
+		Name:     utils.String(name),
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	zones := zones.Expand(d.Get("zones").(*schema.Set).List())
+	if len(zones) > 0 {
+		parameters.Zones = &zones
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, parameters); err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceCapacityReservationGroupRead(d, meta)
+}
+
+func resourceCapacityReservationGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationGroupsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CapacityReservationGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Capacity Reservation Group %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	d.Set("zones", utils.FlattenStringSlice(resp.Zones))
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceCapacityReservationGroupUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationGroupsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CapacityReservationGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	parameters := compute.CapacityReservationGroupUpdate{}
+
+	if d.HasChange("tags") {
+		parameters.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.Name, parameters); err != nil {
+		return fmt.Errorf("updating %s: %+v", id, err)
+	}
+	return resourceCapacityReservationGroupRead(d, meta)
+}
+
+func resourceCapacityReservationGroupDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationGroupsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CapacityReservationGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
+		return fmt.Errorf("deleting %s: %+v", id, err)
+	}
```

Comment:
This api only has the non-future version: https://github.com/Azure/azure-sdk-for-go/blob/bf144f61fde04d2f29dd52d84f25e931cea67fd8/services/compute/mgmt/2021-11-01/compute/capacityreservationgroups.go#L120.
The api issue is that although capacity reservation is deleted and not present in the Get response, capacity reservation group still need some additional wait (about 5 seconds) to realize that, so added the wait in the capacity reservation deletion.

Reply:
i don't see any wait? we souldn't be waiting and instead poll to check until its finished

Reply:
Hi @katbyte The deletion for this resource capacity_reservation_group is synchrounous per [sdk](https://github.com/Azure/azure-sdk-for-go/blob/bf144f61fde04d2f29dd52d84f25e931cea67fd8/services/compute/mgmt/2021-11-01/compute/capacityreservationgroups.go#L120).

And the wait and poll is for the other resource [capaicty_reservation](https://github.com/hashicorp/terraform-provider-azurerm/pull/16464/files#diff-dfef494ce3457b170fb0b57f616090948c20396c11d601436f13b280771ca947R219-R223) below in this change, where the asynchronous sdk has bug

Reply:
I just took another thought. The actual issue https://github.com/Azure/azure-rest-api-specs/issues/18767 happens within deleting the parent resource `capacity_reservation_group` due to cache refresh issue. So instead of adding a wait on `capacity_reservation`, probably it's better to add a retry logic on the parent resource here itself. I'll update the pr and let you know once it's ready to be reviewed again